libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/InsightDrillDialog/DrillDialogInsight.tsx

```diff
@@ -0,0 +1,167 @@
+import { useDashboardComponentsContext } from "../../../../dashboardContexts";
+import {
+    useDashboardSelector,
+    selectColorPalette,
+    selectLocale,
+    selectMapboxToken,
+    selectSeparators,
+    selectSettings,
+    selectIsExport,
+    selectDrillableItems,
+} from "../../../../../model";
+import { IDashboardInsightProps } from "../../types";
+import { useWidgetFiltersQuery } from "../../../common";
+import { useResolveDashboardInsightProperties } from "../useResolveDashboardInsightProperties";
+import { useDrillDialogInsightDrills } from "./useDrillDialogInsightDrills";
+
+const insightStyle: CSSProperties = { width: "100%", height: "100%", position: "relative", flex: "1 1 auto" };
+
+const selectCommonDashboardInsightProps = createSelector(
+    [selectLocale, selectSettings, selectColorPalette],
+    (locale, settings, colorPalette) => ({
+        locale,
+        settings,
+        colorPalette,
+    }),
+);
+
+const selectChartConfig = createSelector(
+    [selectMapboxToken, selectSeparators, selectDrillableItems, selectIsExport],
+    (mapboxToken, separators, drillableItems, isExport) => ({
+        mapboxToken,
+        separators,
+        forceDisableDrillOnAxes: !drillableItems, // to keep in line with KD, enable axes drilling only if using explicit drills
+        isExportMode: isExport,
+    }),
+);
+
+/**
+ * @internal
+ */
+export const DrillDialogInsight = (props: IDashboardInsightProps): JSX.Element => {
+    const {
+        insight,
+        widget,
+        clientHeight,
+        backend,
+        workspace,
+        onError,
+        onDrill: onDrillFn,
+        ErrorComponent: CustomErrorComponent,
+        LoadingComponent: CustomLoadingComponent,
+    } = props;
+
+    // Custom components
+    const { ErrorComponent, LoadingComponent } = useDashboardComponentsContext({
+        ErrorComponent: CustomErrorComponent,
+        LoadingComponent: CustomLoadingComponent,
+    });
+
+    // Context
+    const effectiveBackend = useBackendStrict(backend);
+    const effectiveWorkspace = useWorkspaceStrict(workspace);
+
+    // State props
+    const { locale, settings, colorPalette } = useDashboardSelector(selectCommonDashboardInsightProps);
+    const chartConfig = useDashboardSelector(selectChartConfig);
+
+    // Loading and rendering
+    const [isVisualizationLoading, setIsVisualizationLoading] = useState(false);
+    const [visualizationError, setVisualizationError] = useState<GoodDataSdkError | undefined>();
+
+    const handleLoadingChanged = useCallback<OnLoadingChanged>(({ isLoading }) => {
+        setIsVisualizationLoading(isLoading);
+    }, []);
+
+    /// Filtering
+    const {
+        result: filtersForInsight,
+        status: filtersStatus,
+        error: filtersError,
+    } = useWidgetFiltersQuery(widget, insight && insightFilters(insight));
+
+    const insightWithAddedFilters = insightSetFilters(insight, filtersForInsight as IFilter[]); // TODO how to type this better?
```

Comment:
The TODO and cast should no longer be needed (I remember removing them recently).